src/main/scala/mesosphere/marathon/api/v2/AppHelpers.scala

```diff
@@ -36,14 +36,13 @@ object AppHelpers {
     * "update" operation into a "create" operation. This helper func facilitates that.
     */
   def withoutPriorAppDefinition(update: raml.AppUpdate, appId: PathId): raml.App = {
-    val selectedStrategy = AppConversion.ResidencyAndUpgradeStrategy(
-      residency = update.residency.map(Raml.fromRaml(_)),
+    val selectedStrategy = AppConversion.UpgradeStrategyConverter(
```

Comment:
So `UpgradeStrategyConverter` is equivalent to `ResidencyAndUpgradeStrategy`?

Reply:
After residency became a property that depends on existence of persistent volumes, we no longer need to pass it to the ResidencyAndUpgradeStrategy. The name became a bit obsolete, that's why I renamed it.